lib/features/modeling/BpmnUpdater.js

```diff
@@ -341,11 +345,34 @@ BpmnUpdater.prototype.updateParent = function(element, oldParent) {
   this.updateDiParent(di, parentDi);
 };
 
+// Handle embedded labels with defined bounds, to stay compatible with other vendors
+function shapeHasExplicitlyBoundedInternalLabel(shape) {
+  return (is(shape, 'bpmn:Activity'))
+    && getDi(shape).label
+    && getDi(shape).label.bounds != undefined;
+}
 
 BpmnUpdater.prototype.updateBounds = function(shape) {
 
   var di = getDi(shape);
 
+  // We need to adjust internal bounds if they are defined for compatibility reasons
+  if (shapeHasExplicitlyBoundedInternalLabel(shape)) {
+
+    // clear explicit internal bounds on resize
+    if (shape.width != di.bounds.width || shape.height != di.bounds.height) {
+      di.label = this._bpmnFactory.createDiLabel();
```

Comment:
This means that we also drop the colors, don't we? We should make sure no properties beyond bounds are dropped. Let's have a test case for that. I noticed you already provided a BPMN with labelStyle and color:color -> this is a good example to verify correct behaviour.